### PR TITLE
Set group and visibility of reply to that of parent

### PR DESCRIPTION
**Pull request by seanh**
Set the group and visibility of replies to that of their parents, when
replies are first created.

Also prevent the visibility setting of a reply from overwriting the
cached-in-local-storage visibility level. This prevents the following
problem:

- I'm annotating publicly (my cached-in-local-storage visibility
  setting is public)
- I reply to a private annotation
- The visibility of my reply is set to that of its parent, private
- This visibility is then cached in local storage
- I make a new top-level annotation, and now it's defaulting to private
  instead of public, even though the user never did anything to change
  the visibility from public to private, it was done automatically
  because they replied to a private annotation.

Obviously changing the mode from private to public when replying to a
public annotation was also possible.

This means that if the user _does_ change the visibility of a reply, we
_don't_ cache that either:

- I'm annotating publicly (my cached-in-local-storage visibility
  setting is public)
- I reply to a public annotation
- The visibility of my reply is set to that of its parent, public
- This visibility is not cached in local storage because it's a reply
  (and the value cached in the local storage is already the same anyway)
- I then deliberately change the visibility of my reply from public to
  private
- Again this visibility is not cached in local storage because it's a reply
- The next time I make a new top-level annotation, it will still be
  defaulting to public, will not have changed to private mode

### CODE REVIEW

**Comment by seanh** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;YJNH2Lop&#x2F;143-when-i-reply-to-an-annotation-my-reply-should-go-to-the-same-group-and-have-the-same-visibility"> When I reply to an annotation, my reply should go to the same group and have the same visibility</a>

**Comment by seanh** [comment]
This is done, who wants to review it? @nickstenning @robertknight

**Comment by seanh** [comment]
This should also fix the problem with replying in the stream. No matter what group you had focused last time you used the sidebar, replies in the stream should get the same group as their parents

**Comment by nickstenning** [comment]
This looks great. Nice and clean. One minor thing: the `publish-annotation-btn` directive no longer needs an `isNew` param, so that can be removed (from the directive, its tests, and `annotation.html`).

File: h/static/scripts/directive/test/annotation-test.coffee [hunk 1]
```diff
@@ -691,6 +751,44 @@ describe("AnnotationController", ->
     it("creates the directive without crashing", ->
       createAnnotationDirective({})
     )
+
+    it("sets the permissions of new annotations from local storage", ->
```
Comment (robertknight) [inline h/static/scripts/directive/test/annotation-test.coffee]: here are only two cases here so it doesn't make much difference, but in general I prefer that when you have several tests with very similar logic, they be expressed as:

```
var cases = [...]
cases.forEach(function (case) {
it('does something for case ' + case.name, function () {
  // create variant using the case
});
}
```

**Comment by robertknight** [comment]
LGTM - Just the trivial comment nick mentioned.

**Pull request merged**